### PR TITLE
Improve go.mod workflow and clean up Makefile

### PR DESCRIPTION
**Pull request by Hexcles**
## Review Information

Please see each individual commit message, and skip the second one (it's a mechanic change that will be automated soon).

All tests should pass.

### CODE REVIEW

**Comment by foolip** [comment]
> All tests should pass.

Travis disagrees :)

File: Makefile [hunk 1]
```diff
@@ -37,14 +37,14 @@ prepush: VERBOSE := $() # Empty out the verbose flag.
 prepush: go_build go_test lint
 
 python_test: python3 tox
-	cd $(WPTD_PATH)results-processor; tox
+	cd results-processor; tox
```
Comment (foolip) [inline Makefile]: Or tox -c results-processor?

File: Makefile [hunk 2]
```diff
@@ -74,15 +63,15 @@ go_test_tag_lint:
...
 
-go_medium_test: go_build_test dev_appserver_deps
-	cd $(WPTD_PATH); go test -tags=medium $(VERBOSE) $(FLAGS) ./...
+go_medium_test: go_build dev_appserver_deps
+	go test -tags=medium $(VERBOSE) $(FLAGS) ./...
 
 # Use sub-make because otherwise make would only execute the first invocation
 # of _go_webdriver_test. Variables will be passed into sub-make implicitly.
-go_large_test:
+go_large_test: gcc
```
Comment (Hexcles) [inline Makefile]:
```suggestion
go_large_test:
```

File: Makefile [hunk 3]
```diff
@@ -74,15 +63,15 @@ go_test_tag_lint:
 
 go_test: apt-get-gcc go_small_test go_medium_test
```
Comment (Hexcles) [inline Makefile]:
```suggestion
go_test: go_small_test go_medium_test
```

File: Makefile [hunk 4]
```diff
@@ -74,15 +63,15 @@ go_test_tag_lint:
...
 
-go_small_test: go_build_test
-	cd $(WPTD_PATH); go test -tags=small $(VERBOSE) ./...
+go_small_test: go_build
```
Comment (Hexcles) [inline Makefile]:
```suggestion
go_small_test: go_build gcc
```

File: Makefile [hunk 5]
```diff
@@ -74,15 +63,15 @@ go_test_tag_lint:
...
+	go test -tags=small $(VERBOSE) ./...
 
-go_medium_test: go_build_test dev_appserver_deps
-	cd $(WPTD_PATH); go test -tags=medium $(VERBOSE) $(FLAGS) ./...
+go_medium_test: go_build dev_appserver_deps
```
Comment (Hexcles) [inline Makefile]:
```suggestion
go_medium_test: go_build dev_appserver_deps gcc
```

File: Makefile [hunk 6]
```diff
@@ -100,12 +89,12 @@ webdriver_node_deps:
 
 # _go_webdriver_test is not intended to be used directly; use go_firefox_test or
 # go_chrome_test instead.
-_go_webdriver_test: var-BROWSER java go_build_test xvfb geckodriver dev_appserver_deps
+_go_webdriver_test: var-BROWSER java go_build xvfb geckodriver dev_appserver_deps
```
Comment (Hexcles) [inline Makefile]:
```suggestion
_go_webdriver_test: var-BROWSER java go_build xvfb geckodriver dev_appserver_deps gcc
```

**Pull request merged**